pkg/kfapp/gcp/gcp.go

```diff
@@ -704,11 +704,13 @@ func (gcp *Gcp) updateDM(resources kftypesv3.ResourceEnum) error {
 		}
 		dmOperationEntries = append(dmOperationEntries, storageEntry)
 	}
-	dmEntry, err := gcp.updateDeployment(deploymentmanagerService, gcp.kfDef.Name, CONFIG_FILE)
-	if err != nil {
-		return kfapis.NewKfErrorWithMessage(err, fmt.Sprintf("could not update %v", CONFIG_FILE))
+	if _, mainStatErr := os.Stat(path.Join(gcp.kfDef.Spec.AppDir, GCP_CONFIG, CONFIG_FILE)); mainStatErr == nil {
```

Comment:
nit: add a comment explaining the if condition.
Also it would probably be good to check  if the error is something other than file does not exist to return the error rather than treating it as the file not existing.